common/build.gradle

```diff
@@ -156,6 +167,54 @@ task sourcesJar(type: Jar) {
     destinationDir = reporting.file("$project.buildDir/outputs/jar/")
 }
 
+task createCommonCoverageReportTask() {
+    group = "Common Reporting"
+    description = "Generate Coverage Report tasks for every build Flavour"
+
+    def buildTypes = android.buildTypes.collect { type ->
+        type.name
+    }
+    def productFlavors = android.productFlavors.collect { flavor ->
+        flavor.name
+    }
+    // When no product flavors defined, use empty string
+    if (!productFlavors) productFlavors.add('')
+    productFlavors.each { productFlavorName ->
+        buildTypes.each { buildTypeName ->
```

Comment:
Instead of having a nested loop here for build types and flavors, it would be better to just loop over the `build variants` -- that way you don't have nested loop and also don't need a placeholder empty string for flavors.

Reply:
added buildVariants to loop over, have a look.